File: src/main/java/de/ehmkah/projects/imgdiff/ImgDiffAction.java
package de.ehmkah.projects.imgdiff;

import com.intellij.diff.contents.FileContent;
import com.intellij.diff.requests.SimpleDiffRequest;
import com.intellij.diff.tools.util.DiffDataKeys;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.DumbAwareAction;
import org.jetbrains.annotations.NotNull;

import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.ByteArrayInputStream;
import java.io.IOException;

/**
 * @author Michael Krausse (ehmkah)
 */
public class ImgDiffAction extends DumbAwareAction {

    private DiffedImageCreator diffedImageCreator = new DiffedImageCreator();

    @Override
    public void actionPerformed(@NotNull AnActionEvent anActionEvent) {
        try {
            BufferedImage bufferedImage = diffedImageCreator.getDifferenceImage(readImage(0, anActionEvent), readImage(1, anActionEvent));
            new ImdDiffDialogWrapper(bufferedImage).show();
        } catch (Exception e) {
            System.out.println("Bad luck, my friend" + e);
        }
    }

    private BufferedImage readImage(int index, AnActionEvent anActionEvent) throws IOException {
        ByteArrayInputStream inputstream = new ByteArrayInputStream(((FileContent) ((SimpleDiffRequest) anActionEvent.getDataContext().getData(DiffDataKeys.DIFF_REQUEST)).getContents().get(index)).getFile().contentsToByteArray());
        return ImageIO.read(inputstream);
    }

}

File: src/main/kotlin/de/ehmkah/projects/imgdiff/ImgDiffAction.kt
package de.ehmkah.projects.imgdiff

import com.intellij.diff.contents.FileContent
import com.intellij.diff.requests.DiffRequest
import com.intellij.diff.requests.SimpleDiffRequest
import com.intellij.diff.tools.util.DiffDataKeys
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.project.DumbAwareAction

import javax.imageio.ImageIO
import java.awt.image.BufferedImage
import java.io.ByteArrayInputStream
import java.io.IOException

/**
 * @author Michael Krausse (ehmkah)
 */
class ImgDiffAction : DumbAwareAction() {

    private val diffedImageCreator = DiffedImageCreator()

    override fun actionPerformed(anActionEvent: AnActionEvent) {
        try {
            val bufferedImage = diffedImageCreator.getDifferenceImage(readImage(0, anActionEvent), readImage(1, anActionEvent))
            ImdDiffDialogWrapper(bufferedImage).show()
        } catch (e: Exception) {
            println("Bad luck, my friend$e")
        }

    }

    @Throws(IOException::class)
    private fun readImage(index: Int, anActionEvent: AnActionEvent): BufferedImage {
        val inputstream = ByteArrayInputStream(((anActionEvent.dataContext.getData<DiffRequest>(DiffDataKeys.DIFF_REQUEST) as SimpleDiffRequest).contents[index] as FileContent).file.contentsToByteArray())
        return ImageIO.read(inputstream)
    }

}
